src/lib/json-rpc-router.ts

```diff
@@ -1,12 +1,21 @@
-import type { Dwn } from '@tbd54566975/dwn-sdk-js';
+import type { Dwn, EventSubscriptionHandler } from '@tbd54566975/dwn-sdk-js';
 
 import type { Readable } from 'node:stream';
 
-import type { JsonRpcRequest, JsonRpcResponse } from './json-rpc.js';
+import type { JsonRpcId, JsonRpcRequest, JsonRpcResponse } from './json-rpc.js';
+import type { SocketConnection } from '../connection/socket-connection.js';
 
 export type RequestContext = {
-  dwn: Dwn;
   transport: 'http' | 'ws';
+  dwn: Dwn;
+  socketConnection?: SocketConnection;
+  subscriptionRequest?: {
+    /** The JsonRpcId of the subscription handler */
+    id: JsonRpcId; 
+    /** The `MessageEvent` handler associated with a subscription request, only used in `ws` requests */
+    subscriptionHandler: EventSubscriptionHandler;
+  }
```

Comment:
Design: I am not liking the usage/extension of the `RequestContext`, especially the special casing of `subscriptionRequest` in `buildRequestContext()`.

I think the code could be cleaner if we have dedicated `SocketRequestContext`, and we try to move the subscription specific code into the handler itself. The RPC request routing is a bit unwieldy atm as you probably thought too, but I am fine if you prefer to defer address this in a later PR.

Reply:
Yes I actually agree with you here and was thinking similar, wasn't sure if I wanted to do that in this PR or a separate. Potentially having an `onSubscribe` with a method name and a cleaner context specific to subscription requests.

Was part of the idea behind having `rpc.subscribe` as a prefix, so that the subscription handler can be different and treated like a JSON RPC extension.